libraries/abstractions/ble_hal/test/src/iot_test_ble_hal_common.c

```diff
@@ -1804,49 +1812,53 @@ BTStatus_t IotTestBleHal_WaitEventFromQueue( BLEHALEventsTypes_t xEventName,
             xEventName,
             ( uint32_t ) lhandle );
 
-    pvPtr = checkQueueForEvent( xEventName, lhandle );
-
-    /* If event is not waiting then wait for it. */
-    if( pvPtr == NULL )
+    do
     {
-        do
+        pvPtr = checkQueueForEvent( xEventName, lhandle );
```

Comment:
You can avoid two loops if you pass pxMatch to checkQueueForEvent and check for match inside the function?